src/Browser.php

```diff
@@ -335,4 +337,64 @@ public function withOptions(array $options)
 
         return $browser;
     }
+
+    public function withTimeout($timeout)
+    {
+        $browser = clone $this;
+        $browser->transaction = $this->transaction->withOptions(array(
+            'timeout' => $timeout,
+        ));
+
+        return $browser;
+    }
+
+    public function withoutTimeout()
+    {
+        $browser = clone $this;
+        $browser->transaction = $this->transaction->withOptions(array(
+            'timeout' => null,
+        ));
+
+        return $browser;
+    }
+
+    public function withFollowRedirects($followRedirects)
+    {
+        $browser = clone $this;
+        $browser->transaction = $this->transaction->withOptions(array(
+            'followRedirects' => $followRedirects,
+        ));
+
+        return $browser;
+    }
+
+    public function withMaxRedirects($maxRedirects)
+    {
+        $browser = clone $this;
+        $browser->transaction = $this->transaction->withOptions(array(
+            'maxRedirects' => $maxRedirects,
+        ));
+
+        return $browser;
+    }
+
+    public function withObeySuccessCode($obeySuccessCode)
+    {
+        $browser = clone $this;
+        $browser->transaction = $this->transaction->withOptions(array(
+            'obeySuccessCode' => $obeySuccessCode,
+        ));
+
+        return $browser;
+    }
+
+    public function withStreaming($streaming)
```

Comment:
Makes sense to have this method from a consistency perspective, but I'd rather get rid of the underlying option. I'm currently working on a dedicated PR to add a new `request()` and `requestStreaming()` method which I believe to be a cleaner solution to this :+1:

Reply:
As discussed on Gitter I'll be dropping this assuming your PR doing to refactor gets merged first keeping everything working

Reply:
Removed via https://github.com/clue/reactphp-buzz/pull/170, currently taking a look at rebasing your work, adding documentation and improving API consistency :+1:

Reply:
@clue go for it :+1: